tensorflow/core/platform/file_system.h

```diff
@@ -243,18 +261,21 @@ class FileSystem {
   ///  * OK - successfully created the directory and sub directories, even if
   ///         they were already created.
   ///  * PERMISSION_DENIED - dirname or some subdirectory is not writable.
-  virtual tensorflow::Status RecursivelyCreateDir(const string& dirname);
+  virtual tensorflow::Status RecursivelyCreateDir(const string& dirname) {
+    return RecursivelyCreateDir(dirname, nullptr);
+  }
 
   virtual tensorflow::Status RecursivelyCreateDir(const string& dirname,
-                                                  TransactionToken* token) {
-    return Status::OK();
-  }
+                                                  TransactionToken* token);
 
   /// \brief Deletes the specified directory.
-  virtual tensorflow::Status DeleteDir(const string& dirname) = 0;
+  virtual tensorflow::Status DeleteDir(const string& dirname) {
+    return DeleteDir(dirname, nullptr);
+  };
 
   virtual tensorflow::Status DeleteDir(const string& dirname,
                                        TransactionToken* token) {
+    ;
```

Comment:
```suggestion
```

Reply:
Yes. This is what I meant in the body of the PR. If I do it now, I need to change all filesystems at once since they don't implement transaction API yet.